packages/network/src/logic/Node.ts

```diff
@@ -525,12 +526,21 @@ export class Node extends EventEmitter {
             })
     }
 
-    private connectToBootstrapTrackers(): void {
-        this.trackerRegistry.getAllTrackers().forEach((trackerInfo) => {
-            this.trackerNode.connectToTracker(trackerInfo.ws, PeerInfo.newTracker(trackerInfo.id))
-                .catch((err) => {
-                    this.logger.warn('could not connect to tracker %s, reason: %j', trackerInfo.ws, err)
-                })
+    private maintainTrackerConnections(): void {
+        const activeTrackers = new Set<string>()
```

Comment:
what happens if `maintainTrackerConnections` is called while `connectToTracker` or `disconnectFromTracker` is still pending from previous `setInterval`?

is it safe to call `connectToTracker` twice with the same info? or `disconnectFromTracker` while it's currently trying to `connectToTracker`? 

Maybe this should be a `setTimeout(maintainTrackerConnections, interval)` after a `Promise.allSettled`

Reply:
> what happens if `maintainTrackerConnections` is called while `connectToTracker` or `disconnectFromTracker` is still pending from previous `setInterval`?

It should work fine. The Promise returned by `connectToTracker` is optional to `await` on. It's more unit / integration test purposes where you are interested for a connection to be formed before connecting. Generally outside testing, it's more of a fire-and-forget command style ordeal. 

> is it safe to call `connectToTracker` twice with the same info? or `disconnectFromTracker` while it's currently trying to `connectToTracker`?

Yeah if you call `connectToTracker` twice, the 2nd call is effectively a NOP. There may be a bug now that I read the code wherein a _pending_ connection doesn't properly get killed when calling `disconnectFromTracker`. That should be killed, however.

> Maybe this should be a `setTimeout(maintainTrackerConnections, interval)` after a `Promise.allSettled`
I think we should not rather return a Promise at all (to avoid confusion) and just have it as fire-and-forget style. This way the logic layer is better separated from whatever the connection layer is doing (closer to the connection-manager style of thinking).